instrumentation_http.go

```diff
@@ -218,6 +218,10 @@ type statusCodeRecorder struct {
 	Status int
 }
 
+func (rec *statusCodeRecorder) SetStatus(status int) {
+	rec.Status = status
+}
+
 func (rec *statusCodeRecorder) WriteHeader(status int) {
 	rec.Status = status
```

Comment:
Shall we `SetStatus()` here?

Reply:
In this case, it is a matter of taste. Regarding the `SetStatus` internals and assuming that it will be inlined here. So I am fine with both ways of using it and will update `WriteHeader` and `Write` functions.

Reply:
Let's update them all 👍  In this case it's unlikely we add more functionality to the setter, but generally speaking I think it's good to use a consistent way of accessing the value.